menelaus/injection/class_manipulation.py

```diff
@@ -94,3 +96,119 @@ def class_join(data, target_col, class_1, class_2, new_class, from_index, to_ind
         class_idx = class_idx[(class_idx < to_index) & (class_idx >= from_index)]
         ret[class_idx, target_col] = new_class
     return ret
+
+# endregion
+
+
+# region - LTF-inspired class manipulation functions
+
+def class_probability_shift(data, target_col, from_index, to_index, classes, p_classes):
+    """
+    Resamples the data over a specified window, with altered probability
+    for specified classes (and uniform probability for remaining classes).
+    Accepts ``pandas.DataFrame`` with column names or ``numpy.ndarray``
+    with column indices.
+
+    Ref. :cite:t:`LTFmethods`
+
+    Args:
+        data (np.array): data to inject with drift
+        target_col (int or str): column index/label of targets column
+        from_index: row index at which to start class swap
+        to_index: row index at which to end (non-inclusive) class swap
+        classes (list[int]): list of classes with altered sampling chance
+        p_classes (float): altered sampling chance for members of ``classes``
+
+    Returns:
+        np.array or pd.DataFrame: copy of data, resampled with shifted
+            class probability for 1 or more desired classes
+    """
+    ret = np.copy(data)
+    classes = np.unique(ret[:, target_col])
+
+    # distribution for each data point, and reordering of each point by class
+    p_distribution = np.array([])
+    sample_grouped = np.array([])
+
+    # locate all classes to shift
+    shifted_classes_idx = np.where(data[:, target_col].isin(classes))[0]
+
+    # locate each class in window
+    for cls in classes:
+        cls_idx = np.where(data[:, target_col] == cls)[0]
+        cls_idx = cls_idx[
+            (cls_idx < to_index) &
+            (cls_idx >= from_index)
+        ]
+
+        if cls in classes:
```

Comment:
are you meaning to reference the classes variable defined in L127? I think it should be checking if cls in the classes from the input parameter?

Reply:
Oops, yes, this should iterate over all classes, and then the `if` statement should refer to the shifted classes, but refers to the modified variable instead.